File: mvvm-java/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/common/Event.java
package com.wasisto.githubuserfinder.android.ui.common;

public class Event<T> {

    private T content;

    private boolean hasBeenHandled = false;

    public Event(T content) {
        this.content = content;
    }

    public T getContentIfNotHandled() {
        if (hasBeenHandled) {
            return null;
        } else {
            hasBeenHandled = true;
            return content;
        }
    }

    public T peekContent() {
        return content;
    }

    public boolean hasBeenHandled() {
        return hasBeenHandled;
    }
}

File: mvvm-kotlin/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/common/Event.kt
package com.wasisto.githubuserfinder.android.ui.common

class Event<out T>(private val content: T) {

    var hasBeenHandled = false
        private set

    fun getContentIfNotHandled() = if (hasBeenHandled) null else content.also { hasBeenHandled = true }

    fun peekContent() = content
}